### PR TITLE
Reduce flakiness of end-to-end tests in CI

### PR DESCRIPTION
**Pull request by ericcornelissen**
- Use `--runInBand` when running integration tests in the CI ([source](https://github.com/smooth-code/jest-puppeteer#running-puppeteer-in-ci-environments)).
- Use `jest.retryTimes(3)` to retry failing integration tests 3 times ([source](https://jestjs.io/docs/jest-object#jestretrytimes), thanks for the suggestion @adamrusted).
  - This required upgrading to Jest v27 (c79f1c0b99a7f86958e20fe93bbd756fcfe435a1), as it introduced `jest-circus` as the default test runner.
- Use Jest's fake timers to prevent open handles when test suites finish ([source](https://jestjs.io/docs/timer-mocks), found using `--detectOpenHandles`).

##### Miscellaneous changes

- Rename "integration" tests to "end-to-end" tests.
- Remove [unused variable in `search.test.js`](https://github.com/simple-icons/simple-icons-website/blob/1fb1f1720e7013cf0148895018c0b9d4458170e8/tests/search.test.js#L19).

### CODE REVIEW

**Review by adamrusted — APPROVED** [review]
Thanks for requesting my review, @ericcornelissen - and sorry it's taken so long for me to get around to this. The tests all seem to work properly for me locally - so I'm happy for this to be merged in. Could another @simple-icons/maintainers give this a once over just to be sure we're all happy with it before it gets merged in?

**Comment by ericcornelissen** [comment]
I'm merging this in with just the one approval now in an effort to prevent flaky tests from affecting contributions from outside of the simple-icons organization. If anyone still has suggestions, feel free to share them here or open a separate issue/Pull Request for it :slightly_smiling_face:

**Pull request merged**
